backend/muster-roll-service/src/main/java/org/egov/validator/MusterRollValidator.java

```diff
@@ -1,9 +1,284 @@
+            throw new CustomException("MUSTER_ROLL_SEARCH_CRITERIA_REQUEST", " Muster roll search criteria request is mandatory");
+        }
+        if (StringUtils.isBlank(searchCriteria.getTenantId())) {
+            throw new CustomException("TENANT_ID", "Tenant is mandatory");
+        }
+    }
+
+
+    private void validateRequestInfo(RequestInfo requestInfo, Map<String, String> errorMap) {
+        if (requestInfo == null) {
+            throw new CustomException("REQUEST_INFO", "Request info is mandatory");
+        }
+        if (requestInfo.getUserInfo() == null) {
+            throw new CustomException("USERINFO", "UserInfo is mandatory");
+        }
+        if (requestInfo.getUserInfo() != null && StringUtils.isBlank(requestInfo.getUserInfo().getUuid())) {
+            throw new CustomException("USERINFO_UUID", "UUID is mandatory");
+        }
+    }
+
+    private void validateCreateMusterRoll(MusterRoll musterRoll, Map<String, String> errorMap) {
+        if (musterRoll == null) {
+            throw new CustomException("MUSTER_ROLL","Muster roll is mandatory");
+        }
+        if (musterRoll.getTenantId() == null) {
+            throw new CustomException("TENANT_ID","TenantId is mandatory");
+        }
+        if (musterRoll.getRegisterId() == null) {
+            throw new CustomException("REGISTER_ID","RegisterId is mandatory");
+        }
+        if (musterRoll.getStartDate() == null) {
+            throw new CustomException("START_DATE_EMPTY","StartDate is mandatory");
+        }
+
+        //Check if the startDate is Monday - UI sends the epoch time in IST
+        LocalDate startDate = Instant.ofEpochMilli(musterRoll.getStartDate()).atZone(ZoneId.of(/*serviceConfiguration.getTimeZone()*/"Asia/Kolkata")).toLocalDate();
+        if (startDate.getDayOfWeek() != DayOfWeek.MONDAY) {
+            throw new CustomException("START_DATE_MONDAY","StartDate should be Monday");
+        }
+
+        log.info("MusterRollValidator::validateCreateMusterRoll::startDate in epoch::"+musterRoll.getStartDate());
+        log.info("MusterRollValidator::validateCreateMusterRoll::startDate::"+startDate);
+
+        //Set the endDate as SUNDAY
+        LocalDate endDate = startDate.plusDays(6);
+        musterRoll.setEndDate(endDate.atStartOfDay(ZoneId.of(/*serviceConfiguration.getTimeZone()*/"Asia/Kolkata")).toInstant().toEpochMilli());
+
+    }
+
+    private void validateUpdateMusterRoll(MusterRoll musterRoll, RequestInfo requestInfo, Workflow workflow, Map<String, String> errorMap) {
+        if (musterRoll == null) {
+            throw new CustomException("MUSTER_ROLL","Muster roll is mandatory");
+        }
+        if (musterRoll.getTenantId() == null) {
+            throw new CustomException("TENANT_ID","TenantId is mandatory");
+        }
+        if (musterRoll.getId() == null) {
+            throw new CustomException("MUSTER_ROLL_ID","MusterRollId is mandatory");
+        }
+        if (musterRoll.getMusterRollNumber() == null) {
```

Comment:
The record to be updated is chosen based on the id of the muster roll. We do not mandatorily need muster roll number. 
Even if we get that in the update request, we will not update the muster roll number.

Reply:
The muster roll number was required for the workflow service. I realized that it can fetched from the existing record. So removed this validation.